pulsar-client/src/main/java/org/apache/pulsar/client/impl/AbstractBatchMessageContainer.java

```diff
@@ -37,11 +37,11 @@
     protected ProducerImpl producer;
 
     protected int maxNumMessagesInBatch;
+    protected int maxBytesInBatch;
     protected int numMessagesInBatch = 0;
     protected long currentBatchSizeBytes = 0;
 
-    protected static final int INITIAL_BATCH_BUFFER_SIZE = 1024;
-    protected static final int MAX_MESSAGE_BATCH_SIZE_BYTES = 128 * 1024;
+    protected static final int INITIAL_BATCH_BUFFER_SIZE = 128 * 1024;
```

Comment:
It's better to keep INITIAL_BATCH_BUFFER_SIZE as 1KB, in key_based batcher properly create lot of ByteBuf, if the INITIAL_BATCH_BUFFER_SIZE is 128KB, more memory overhead may be incurred.

We can add the option for init batch buffer size in the future, so that user can decide according to their own situation.